src/components/Table/components/SettingsBar/index.tsx

```diff
@@ -31,7 +31,9 @@ export interface SettingsBarProps<D extends object> {
     loading?: boolean;
     gotoPage?: (pageIndex: number) => void;
     previousPage?: () => void;
+    canPreviousPage?: boolean;
     nextPage?: () => void;
+    canNextPage?: boolean;
```

Comment:
Nit: Might be good to add a verb here, like `canNavigateToNextPage` or `canVisitNextPage`?

Reply:
I think this is an internal component. It may be good to keep the consistency of the api of react-table?